pyparsing.py

```diff
@@ -1752,21 +1752,23 @@ def parseString(self, instring, parseAll=False):
         ...
         pyparsing.ParseException: Expected end of text, found 'b'  (at char 5), (line:1, col:6)
         """
-        
+        if parseAll:
```

Comment:
This change is necessary to pass unit tests:

        if parseAll:
            str_end = StringEnd()
            for expr in self.ignoreExprs:
                str_end.ignore(expr)
            return (self + str_end).parseString(instring)

Also, I think this PR should contain only this change and the other changes you've proposed to pyparsing.py - the other changed files should be part of a different PR.

Reply:
No, this is wrong, see more recent comments on the PR.